llvm/lib/TableGen/TGParser.cpp

```diff
@@ -4439,16 +4440,16 @@ bool TGParser::CheckTemplateArgValues(
                "result of template arg value cast has wrong type");
         Value = Value->cloneWithValue(CastValue);
       } else {
-        PrintFatalError(Loc, "Value specified for template argument '" +
-                                 Arg->getNameInitAsString() + "' is of type " +
-                                 ArgValue->getType()->getAsString() +
-                                 "; expected type " + ArgType->getAsString() +
-                                 ": " + ArgValue->getAsString());
+        HasError |= Error(
```

Comment:
The change is good, but can we also adjust the error Loc to be more accurate? It seems that the Loc here is the position of the name of the class. but it might be better to show the location of the actual class arguments? I am not sure if those locations are actually tracked by the TG Lexer/Parser.

Reply:
I've added separate location tracking for the arguments in the parser since the `Init`s sadly do not track locations (since that would make the node folding redundant). Let me know what you think

Reply:
Thanks for adding this. I also created an issue: https://github.com/llvm/llvm-project/issues/121792 to look into tracking locations for Inits.